torch/csrc/api/src/nn/modules/adaptive.cpp

```diff
@@ -61,14 +61,40 @@ void AdaptiveLogSoftmaxWithLossImpl::reset_parameters() {
 }
 
 ASMoutput AdaptiveLogSoftmaxWithLossImpl::forward(const Tensor& input, const Tensor& target) {
-  TORCH_CHECK(input.size(0) == target.size(0),
-      "Input and target should have the same size in the batch dimension.");
+  auto targ_dim = target.dim();
+
+  TORCH_CHECK(
+    targ_dim == 1 || targ_dim == 0,
+    "0D or 1D target tensor expected, multi-target not supported");
+
+  if (targ_dim == 1) {
+    TORCH_CHECK(
+      input.dim() == 2,
+      "1D target tensor expects 2D input tensors, but found inputs with sizes ",
+      input.sizes(),
+      ".");
+  } else {
+    TORCH_CHECK(
+      input.dim() == 1,
+      "0D target tensor expects 1D input tensors, but found inputs with sizes ",
+      input.sizes(),
+      ".");
+  }
+
+  Tensor new_input, new_target;
```

Comment:
nit: the usual style is to use names like `input_` / `target_` for the args. Then you can do:

```cpp
bool is_batched = (target_dim > 0);
Tensor input = is_batched ? input_.unsqueeze(0) : input_;
Tensor target = is_batched ? target_.unsqueeze(0) : target_;
```